app/org/sagebionetworks/bridge/models/surveys/SurveyRule.java

```diff
@@ -98,21 +133,26 @@ public boolean equals(Object obj) {
                Objects.equals(operator, other.operator) &&
                Objects.equals(value, other.value) &&
                Objects.equals(endSurvey, other.endSurvey) &&
-               Objects.equals(assignDataGroup, other.assignDataGroup);
+               Objects.equals(assignDataGroup, other.assignDataGroup) &&
+               Objects.equals(dataGroups, other.dataGroups);
```

Comment:
Added (these are used for tests and SurveyRuleTest was failing... I'm rerunning all the tests before committing next round of changes).